lib/omnibus-ctl.rb

```diff
@@ -176,12 +178,28 @@ def add_command_under_category(name, category, description, arity=1, &block)
       @category_command_map[category] = {} unless @category_command_map.has_key?(category)
       @category_command_map[category][name] = { :desc => description, :arity => arity }
       metaclass = class << self; self; end
-      # Ruby does not like dashes in method names
       method_name = name.gsub(/-/, "_")
       metaclass.send(:define_method, method_name.to_sym) { |*args| block.call(*args) }
     end
 
-    def exit!(error_code)
+    def exit!(error_code, cmd_name = nil)
+      # Rather than force every caller of exit! (which is pretty much
+      # everywhere) to pass in an additional argument, we're going to instead
+      # take a peak - if we recognize the calling method as a valid command,
+      # we're process a post-hook for that command. For those that are indirectly calling
+      # exit! from a common function, they can provide the name.
+      if cmd_name.nil?
+        cmd_name = caller[0][/`.*'/][1..-2]
```

Comment:
If you don't find something better, we should comment on how this expression works.